src/hotspot/share/opto/vectornode.hpp

```diff
@@ -144,6 +144,22 @@ class VectorNode : public TypeNode {
 };
 
 //===========================Vector=ALU=Operations=============================
+// Base IR node for saturating signed / unsigned operations.
+// Saturating operation prevents wrapping result value in over/underflowing
+// scenarios, instead returns delimiting MAX/MIN value of result type.
+class SaturatingVectorNode : public VectorNode {
+ private:
+  const bool _is_unsigned;
+
+ public:
+  SaturatingVectorNode(Node* in1, Node* in2, const TypeVect* vt, bool is_unsigned) : VectorNode(in1, in2, vt), _is_unsigned(is_unsigned) {
+    init_class_id(Class_SaturatingVector);
+  }
+
+  // Needed for proper cloning.
+  virtual uint size_of() const { return sizeof(*this); }
+  bool is_unsigned() { return _is_unsigned; }
```

Comment:
Can you put this in the `print_spec`, so the IR dump shows if it is unsigned?

Reply:
Hi @eme64 , I see print_spec is mainly used for dumping information about VTransformVectorNode,  please note newly added situating operations are not being auto-vectorized and are are emitted through VectorAPI based explicit vectorization flow.

Reply:
I think you meant Node::dump_spec. Its a good suggestions, addressed.